src/Server/LanguageServer.cs

```diff
@@ -209,9 +212,24 @@ public IDisposable AddHandlerMatcher(IHandlerMatcher handlerMatcher)
                 ReferencesProvider = ccp.HasStaticHandler(textDocumentCapabilities.References),
                 RenameProvider = ccp.HasStaticHandler(textDocumentCapabilities.Rename),
                 SignatureHelpProvider = ccp.GetStaticOptions(textDocumentCapabilities.SignatureHelp).Get<ISignatureHelpOptions, SignatureHelpOptions>(SignatureHelpOptions.Of),
-                WorkspaceSymbolProvider = ccp.HasStaticHandler(workspaceCapabilities.Symbol)
+                WorkspaceSymbolProvider = ccp.HasStaticHandler(workspaceCapabilities.Symbol),
+                ImplementationProvider = ccp.GetStaticOptions(textDocumentCapabilities.Implementation).Get<IImplementationOptions, ImplementationOptions>(ImplementationOptions.Of),
+                TypeDefinitionProvider = ccp.GetStaticOptions(textDocumentCapabilities.TypeDefinition).Get<ITypeDefinitionOptions, TypeDefinitionOptions>(TypeDefinitionOptions.Of),
+                ColorProvider = ccp.GetStaticOptions(textDocumentCapabilities.ColorProvider).Get<IColorOptions, StaticColorOptions>(ColorOptions.Of),
             };
 
+            if (_collection.ContainsHandler(typeof(IDidChangeWorkspaceFoldersHandler)))
```

Comment:
@tintoy we can make changes here to help determine if we should support workspace folders, however it's up to the implementor if they support them or not.